### ISSUE TITLE
Simple GitPod Questions

### ISSUE THREAD

**Issue by DaniilNapuda** [gitpod-io/gitpod#1569]
I started using GitPod and don’t understand how:
- Set up a project for C # WinForms
- Compile project
- Make it so that you can connect your namespace from another file

**Comment by Kreyren** [comment]
> Set up a project for C # WinForms @DaniilNapuda

Should work out of the box afaik

> Compile project @DaniilNapuda

Depends on the project, provide source or tell us how do you build it locally.

> Make it so that you can connect your namespace from another file @DaniilNapuda

I don't understand what do you mean O.o elaborate..

**Comment by DaniilNapuda** [comment]
1. I am doing an empty project. I create it on GitHub, add https://gitpod.io/# before the URL and create .gitpod.yml, .gitpod.Dockerfile, and tasks.json.
![image](https://user-images.githubusercontent.com/65886954/83346729-455ec400-a349-11ea-9d92-2bccde717eb9.png)

2. Compiling all of the above

3. Are you not a C # programmer?

**Comment by Kreyren** [comment]
> I am doing an empty project. I create it on GitHub, add gitpod.io/# before the URL and create .gitpod.yml, .gitpod.Dockerfile, and tasks.json. @DaniilNapuda 

So what is the issue?

> Compiling all of the above @DaniilNapuda 

Assuming that being an empty project there is nothing to compile?

If you want proof-of-contept then @JesterOrNot has that on https://gitpod.io/#https://github.com/JesterOrNot/Gitpod-dotnet-Debug 

Note that gitpod has `gitpod/workspace-dotnet` docker environment which may interests you

> Are you not a C # programmer? @DaniilNapuda

I wish i wasn't it's s!@$% lang :p Used to do Skyrim mods it in, but it's too inefficient and limited.
Now i would consider myself understanding the theory, but i wouldn't be comfortable in working with it.

**Comment by Kreyren** [comment]
Referencing relevant environments:
- https://github.com/gitpod-io/workspace-images/blob/master/dotnet/Dockerfile
- https://github.com/gitpod-io/workspace-images/blob/master/dotnet-vnc/Dockerfile

**Comment by DaniilNapuda** [comment]
1. That's why (bottom right corner)
![image](https://user-images.githubusercontent.com/65886954/83347099-9f14bd80-a34c-11ea-83f8-1ce270d016ad.png)

2. The question is to compile .exe, without this:
![Error](https://user-images.githubusercontent.com/65886954/83347155-2feb9900-a34d-11ea-9134-795bcc76cbc0.png)

**Comment by Kreyren** [comment]
> That's why (bottom right corner) @DaniilNapuda

Exit code 127 should be standardized on **command not found** which points to pebcak, check your tasks.json

> The question is to compile .exe, without this: @DaniilNapuda

I believe that i've already answered this in https://github.com/gitpod-io/gitpod/issues/1560#issuecomment-635908919

Did you find it difficult to understand or missed it?

Or am i missunderstanding the issue? O.o

**Comment by DaniilNapuda** [comment]
2. I do not work in Visual Studio

**Comment by Kreyren** [comment]
> I do not work in Visual Studio @DaniilNapuda 

Gitpod provides theia (https://github.com/eclipse-theia/theia) it looks like Visual Studio or Vscode but it's a rewrite.

**Comment by Kreyren** [comment]
Or how is Visual Studio relevant here? O.o

**Comment by DaniilNapuda** [comment]
Is that solution not shown in VS?

**Comment by DaniilNapuda** [comment]
Is there a code that disables CMD and it is not displayed?

**Comment by Kreyren** [comment]
Referencing https://www.mono-project.com/docs/gui/ for GUI toolkits

Assigning @JesterOrNot, because he has more experience with dotnet on gitpod and i currently don't have time to make Proof-of-concept of GUI on gitpod.

**Comment by DaniilNapuda** [comment]
thank

**Comment by JesterOrNot** [comment]
Hi @DaniilNapuda for compiling and running C# In Gitpod please see [the docs](https://www.gitpod.io/docs/languages/dotnet/#installing-the-net-tools) I will create a brief GUI example in Gitpod.

**Comment by JesterOrNot** [comment]
See [this repo](https://github.com/JesterOrNot/Gitpod-Dotnet-GUI) to see a minimalistic setup for compiling and running WinForms in Gitpod

**Comment by JesterOrNot** [comment]
See the following for more info of why this works

- https://www.mono-project.com/docs/gui/winforms/getting-started-guide/
- https://www.mono-project.com/docs/gui/winforms/

**Comment by DaniilNapuda** [comment]
JesterOrNot, anyway, the program compiles with CMD:

![image](https://user-images.githubusercontent.com/65886954/83388611-ef595180-a418-11ea-8a8c-b05c109ecb29.png)

Could this be related to OS settings?

**Comment by Kreyren** [comment]
FWIW it seems to me as a configuration in the code assuming that visual studio on windows (for which these are designed for and for linux which we are using it's allegedly a M$ sabotaged implementation) just provides a front-end to this? There is also a monoeditor (or however they call it) which we may be able to get running on gitpod in the worst case scenario.

**Comment by DaniilNapuda** [comment]
You can write more about monoeditor?

**Comment by Kreyren** [comment]
> You can write more about monoeditor?

Not really i just saw it being mensioned on the mono project while questioning my life decisions to work with dotnet again to make the PoC for gitpod while realizing that i can use Sean to do it instead of me while keeping my sanity from skyrim development-induset PTSD to be honest.

See https://www.monodevelop.com/

**Comment by Kreyren** [comment]
Also FWIW the mentioned repository in https://github.com/gitpod-io/gitpod/issues/1569#issuecomment-636485147 works for me without it opening the unexpected console in gitpod:

![image](https://user-images.githubusercontent.com/11302521/83394189-6df1c700-a3f8-11ea-831e-c5197df06cbe.png)

but i doubt this being related to OS settings aldo possible i guess?

**Comment by DaniilNapuda** [comment]
Not in GitPod, but by launching .exe on a PC.

**Comment by Kreyren** [comment]
> Not in GitPod, but by launching .exe on a PC.

I know, but i can't test dotnet on my system atm so i've tested at least gitpod hoping it would be helpful~

**Comment by DaniilNapuda** [comment]
I realized that I can not configure tasks.json ...
Please, tell me how to do this.

**Comment by Kreyren** [comment]
@DaniilNapuda AFAIK you can save tasks.json in `.vscodetasks.json` or `.theia/tasks.json`

**Comment by DaniilNapuda** [comment]
I need a sample for writing code

**Comment by DaniilNapuda** [comment]
How to save i know

**Comment by Kreyren** [comment]
> I need a sample for writing code @DaniilNapuda 

Something like this? 

https://github.com/dotnet/try/blob/master/.vscode/tasks.json

**Comment by DaniilNapuda** [comment]
Yes, but it doesn’t help :(
![image](https://user-images.githubusercontent.com/65886954/83624624-09cc3000-a5bd-11ea-8c8d-79d94acc784e.png)

**Comment by DaniilNapuda** [comment]
Maybe I'm doing something wrong?

**Comment by Kreyren** [comment]
> Maybe I'm doing something wrong?

Yes :p you can't just copy paste some random `tasks.json` these are expected to be coded for your program.

**Comment by DaniilNapuda** [comment]
Explain to me the algorithm of actions, please.
I just can't understand...

**Comment by JesterOrNot** [comment]
As of right now we can't debug Dotnet apps in Gitpod (as least as far as I know) see https://community.gitpod.io/t/how-to-debug-c-on-gitpod/514/5

**Comment by Kreyren** [comment]
@DaniilNapuda To be honest this seems like dotnet specific questions for which gitpod's tracker is not a best place to ask so note IRC channel on https://github.com/gitpod-io/gitpod/issues/1560#issuecomment-635908919 where they will respond to you much faster and more accurately.

Let me know if you need help with setting up IRC client.

TLDR for freenode registration is:
1. Open an IRC client which is either https://webchat.freenode.net or install `hexchat` https://hexchat.github.io/ (note that the non-windows store version is free)

### For Hexchat
2. Download hexchat
2.1 On Linux hexchat should be packaged by your distribution so you can install it for example using `apt install -y hexchat`
2.2 On windows use https://hexchat.github.io/downloads.html and download windows 7 executable which is free
3. Open hexchat and navigate to Server List
![image](https://user-images.githubusercontent.com/11302521/83670290-39863280-a5d3-11ea-94b6-2af930914e23.png)
4. On server list find Freenode
![image](https://user-images.githubusercontent.com/11302521/83670337-4c990280-a5d3-11ea-8a50-23977894ace0.png)
5. If you are not registered then press **Connect**.
6. Register your account (as explained on https://freenode.net/kb/answer/registration) by writing `/msg NickServ REGISTER YourPassword youremail@example.com` once invoked you will recieve instructions on your specified e-mail.
7. (optional) if you have issue with the registration you can invoke `/join #freenode` and they will help you (don't be afraid to ask questions! :p)
![image](https://user-images.githubusercontent.com/11302521/83670583-b1545d00-a5d3-11ea-95b4-20da8b5ee990.png)
8. Login using `/msg NickServ identify yourPassword` (red) note that your nick (green) has to be the same as the one you registered in step 6.
![image](https://user-images.githubusercontent.com/11302521/83670797-0001f700-a5d4-11ea-850d-de3039779c0a.png)
9. Invoke `/join ##csharp` and ask your questions! ^-^

### For Webchat
2. Choose an unique nickname (red), complete captha (green) and press **START**
![image](https://user-images.githubusercontent.com/11302521/83669048-39853300-a5d1-11ea-99fa-6614dcc22891.png)
3. Register your account (as explained on https://freenode.net/kb/answer/registration) by writing `/msg NickServ REGISTER YourPassword youremail@example.com` once invoked you will recieve instructions on your specified e-mail.
3.1 If your nickname is already taken then invoke `/nick newNickname` and repeat step 3.
![image](https://user-images.githubusercontent.com/11302521/83669547-055e4200-a5d2-11ea-860d-63c324c41136.png)
4. (optional) if you have issue with the registration you can invoke `/join #freenode` and they will help you (don't be afraid to ask questions! :p)
![image](https://user-images.githubusercontent.com/11302521/83669767-62f28e80-a5d2-11ea-8e1f-9ac138626ba1.png)
5. Login using `/msg NickServ identify yourPassword` (red) note that your nick (green) has to be the same as the one you registered in step 3.
![image](https://user-images.githubusercontent.com/11302521/83669966-b533af80-a5d2-11ea-83c1-130f2750053e.png)
6. Write `/join ##csharp` and ask your questions! ^-^

---

> Explain to me the algorithm of actions, please.
> I just can't understand... @DaniilNapuda 

I have no idea what do you mean by 'alhorithm of action'.. The general practice is to make a standard file hierarchy in the target directory (or using custom file hierarchy if standard is not enough for you which is used for very specific projects and should be avoided at all cost otherwise) and if needed set up `tasks.json` so that you can use vscode/theia GUI.

Note https://code.visualstudio.com/docs/editor/tasks for `tasks.json` documentation where afaik theia is using the same syntax.
